cpp/src/binaryop/compiled/binops_custom.cu

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "binops_custom.cuh"
+
+#include <cudf/binaryop.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/iterator.cuh>
+#include <cudf/detail/null_mask.hpp>
+#include <cudf/detail/nvtx/ranges.hpp>
+#include <cudf/scalar/scalar_device_view.cuh>
+#include <cudf/table/table_view.hpp>
+#include <cudf/utilities/span.hpp>
+
+#include <binaryop/jit/operation.hpp>
+
+#include <rmm/device_uvector.hpp>
+
+namespace cudf {
+
+namespace binops {
+namespace compiled {
+// Defined in util.cpp
+data_type get_common_type(data_type out, data_type lhs, data_type rhs);
+bool is_supported_operation(data_type out, data_type lhs, data_type rhs, binary_operator op);
```

Comment:
Make a header for `util.cpp` instead of manually forward declaring.

Reply:
moved to `src/binaryop/compiled/binary_ops.hpp`